test/functional/wallet_upgradewallet.py

```diff
@@ -137,5 +183,165 @@ def run_test(self):
         # after conversion master key hash should be present
         assert_is_hex_string(wallet.getwalletinfo()['hdseedid'])
 
+        self.log.info('Intermediary versions don\'t effect anything')
+        copy_non_hd()
+        # Wallet starts with 60000
+        assert_equal(60000, wallet.getwalletinfo()['walletversion'])
+        wallet.unloadwallet()
+        before_checksum = sha256sum_file(node_master_wallet)
+        node_master.loadwallet('')
+        # Can "upgrade" to 129999 which should have no effect on the wallet
+        wallet.upgradewallet(129999)
+        assert_equal(60000, wallet.getwalletinfo()['walletversion'])
+        wallet.unloadwallet()
+        assert_equal(before_checksum, sha256sum_file(node_master_wallet))
+        node_master.loadwallet('')
+
+        self.log.info('Wallets cannot be downgraded')
+        copy_non_hd()
+        assert_raises_rpc_error(-4, 'Cannot downgrade wallet', wallet.upgradewallet, 40000)
+        wallet.unloadwallet()
+        assert_equal(before_checksum, sha256sum_file(node_master_wallet))
+        node_master.loadwallet('')
+
+        self.log.info('Can upgrade to HD')
+        # Inspect the old wallet and make sure there is no hdchain
+        orig_kvs = dump_bdb_kv(node_master_wallet)
+        assert b'\x07hdchain' not in orig_kvs
+        # Upgrade to HD, no split
+        wallet.upgradewallet(130000)
+        assert_equal(130000, wallet.getwalletinfo()['walletversion'])
+        # Check that there is now a hd chain and it is version 1, no internal chain counter
+        new_kvs = dump_bdb_kv(node_master_wallet)
+        assert b'\x07hdchain' in new_kvs
+        hd_chain = new_kvs[b'\x07hdchain']
+        assert_equal(28, len(hd_chain))
+        hd_chain_version, external_counter, seed_id = struct.unpack('<iI20s', hd_chain)
+        assert_equal(1, hd_chain_version)
+        seed_id = bytearray(seed_id)
+        seed_id.reverse()
+        old_kvs = new_kvs
+        # First 2 keys should still be non-HD
+        for i in range(0, 2):
+            info = wallet.getaddressinfo(wallet.getnewaddress())
+            assert 'hdkeypath' not in info
+            assert 'hdseedid' not in info
+        # Next key should be HD
+        info = wallet.getaddressinfo(wallet.getnewaddress())
+        assert_equal(binascii.hexlify(seed_id).decode(), info['hdseedid'])
```

Comment:
in commit 91125ca0d459c112e48db9bbaa10dd1380a3e32c:

Pretty sure you can just write seed_id.hex() (without binascii)

Reply:
Done.